src/modules/replay/Replay.cpp

```diff
@@ -387,7 +387,7 @@ Replay::readAndAddSubscription(std::ifstream &file, uint16_t msg_size)
 	_subscription_file_pos = file.tellg();
 
 	uint8_t multi_id = *(uint8_t *)message;
-	uint16_t msg_id = ((uint16_t) message[1]) | (((uint16_t) message[2]) << 8);
+	uint16_t msg_id = static_cast<unsigned char>(message[1]) + (static_cast<unsigned char>(message[2]) << 8);
```

Comment:
What about not using a `char*` in the first place (`char *message` on line 374)? The underlying data is uint8_t.

``` C++
uint8_t *message = _read_buffer.data();
```

Reply:
Sounds good. I will test that.

Reply:
Tested and still works with IDs >= 128. Force-pushed this.